routers/api/v1/repo/mirror.go

```diff
@@ -63,3 +74,302 @@ func MirrorSync(ctx *context.APIContext) {
+	pushMirrors, err := repo_model.GetPushMirrorsByRepoID(repo.ID, db.ListOptions{})
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "PushMirrorSync", err)
+		return
+	}
+	for _, mirror := range pushMirrors {
+		mirror_service.SyncPushMirror(ctx, mirror.ID)
+	}
+
+	ctx.Status(http.StatusOK)
+}
+
+// ListPushMirrors get list of push mirrors of a repository
+func ListPushMirrors(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/push_mirrors repository repoListPushMirrors
+	// ---
+	// summary: Get all push mirrors of the repository
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: limit
+	//   in: query
+	//   description: page size of results
+	//   type: integer
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/PushMirrorList"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	repo := ctx.Repo.Repository
+	listOptions := utils.GetListOptions(ctx)
+	if !setting.Mirror.Enabled {
+		ctx.Error(http.StatusBadRequest, "ListPushMirrors", "Mirror feature is disabled")
+		return
+	}
+	// Get All push mirrors of a specific repo
+	pushMirrors, err := repo_model.GetPushMirrorsByRepoID(repo.ID, listOptions)
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "ListPushMirrors", err)
+		return
+	}
+
+	responsePushMirrors := make([]*api.PushMirror, 0)
+	for _, mirror := range pushMirrors {
+		responsePushMirrors = append(responsePushMirrors, convert.ToPushMirror(mirror, repo))
+	}
+	ctx.SetLinkHeader(len(pushMirrors), listOptions.PageSize)
+	ctx.SetTotalCountHeader(int64(len(pushMirrors)))
+	ctx.JSON(http.StatusOK, responsePushMirrors)
+}
+
+// GetPushMirrorByName get push mirror of a repository by name
+func GetPushMirrorByName(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/push_mirrors/{name} repository repoGetPushMirrorByID
+	// ---
+	// summary: Get push mirror of the repository by ID
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo to sync
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo to sync
+	//   type: string
+	//   required: true
+	// - name: name
+	//   in: path
+	//   description: remote name of push mirror
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/PushMirror"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	repo := ctx.Repo.Repository
+	id := ctx.Params(":name")
```

Comment:
To avoid confusion this variable is better to be named `name` or `mirrorName` or something alike.